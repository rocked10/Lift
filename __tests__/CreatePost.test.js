import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CreatePost from '../screens/createPost';
import { render, fireEvent, waitFor } from "@testing-library/react-native";

describe('Testing Create Post Screen', () => {
    const navigate = jest.fn(); 
    const wrapper = renderer.create(<CreatePost navigation={ navigate } route={{params: {name: 'jeremy', role:'coach', userId: '123'}}} />);

    it('Should render', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it ("Renders elements", () => {   
        const navigate = jest.fn(); 
        const { getByTestId, getByPlaceholderText,} = render(<CreatePost navigation={ navigate } route={{params: {name: 'jeremy', role:'coach', userId: '123'}}} />);
        // getByTestId('Profile Card'); this fails?
        getByPlaceholderText("What's your workout of the day?");
        getByTestId('Attach file icon');
    });
});