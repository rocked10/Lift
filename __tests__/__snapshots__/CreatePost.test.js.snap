// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Create Post Screen Should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 4,
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            Object {
              "minHeight": 72,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 40,
                "justifyContent": "center",
                "marginRight": 16,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <View
            color="white"
            style={
              Array [
                Object {
                  "backgroundColor": "#6200ee",
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <Text />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "500",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "minHeight": 30,
                      },
                      Object {
                        "marginBottom": 0,
                      },
                      Object {
                        "fontFamily": "lato-bold",
                      },
                    ],
                  ],
                ],
              ]
            }
          >
            jeremy
          </Text>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "marginVertical": 0,
                        "minHeight": 20,
                      },
                      undefined,
                    ],
                  ],
                ],
              ]
            }
          >
            coach
          </Text>
        </View>
        <View>
          <View
            style={
              Object {
                "backgroundColor": "#6200ee",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 2,
                "marginRight": 14,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0.75,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Post
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 8,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      defaultValue=""
      multiline={true}
      onChangeText={[Function]}
      placeholder="What's your workout of the day?"
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
          "fontFamily": "lato-regular",
          "fontSize": 18,
          "padding": 12,
        }
      }
      textAlignVertical="top"
      underlineColorAndroid="transparent"
    />
    <Text />
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "flexDirection": "row",
        "margin": 12,
        "position": "absolute",
      }
    }
  />
</View>
`;
